### ISSUE TITLE
Add support for multiple xmlns pointing to the same uri

### ISSUE THREAD

**Issue by XiaotianNetlift** [Xavalon/XamlStyler#172]
When formatting a document with multiple xmlns are pointing to the same uri, every `XName` is replaced by the one prefixed with the xmlns declared the last.

Given this xaml:
```xaml
<Window x:Class="MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:not_win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <StackPanel>
    <win:TextBlock Text="Windows" />
    <not_win:TextBlock Text="other platform..." />
  </StackPanel>
</Window>
```

Expected formatted xaml:
```xaml
<Window x:Class="MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:not_win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <StackPanel>
    <win:TextBlock Text="Windows" />
    <not_win:TextBlock Text="other platform..." />
  </StackPanel>
</Window>
```

Actual result:
```xaml
<win:Window x:Class="MainWindow"
            xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
            xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
            xmlns:not_win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
            xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
  <win:StackPanel>
    <win:TextBlock Text="Windows" />
    <win:TextBlock Text="other platform..." />
  </win:StackPanel>
</win:Window>
```

**Comment by MatFillion** [comment]
I've noticed the same problem.

**Comment by MatFillion** [comment]
@grochocki have you had the chance to look at the proposed PR ?  https://github.com/Xavalon/XamlStyler/pull/173

**Comment by grochocki** [comment]
I'll be able to take a look tomorrow. Thanks for the remidner!

**Comment by grochocki** [comment]
Fixed via #173

**Comment by artemious7** [comment]
I experience the very same bug in XAMLStyler v3.2.5 from the MarketPlace. Looks like it was **not fixed**.

If I try to use this workaround
`xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation+asdasd"`
(putting `+asdasd` in the end)
the XamlStyler won't put the `win:` namespace everywhere, but `win:` will raise an error in the Xaml Designer because of the unknown namespace and it won't build.

**It's not a solution** to add and then remove `+asdasd` every time I want to format a document!

**Comment by artemious7** [comment]
@grochocki please help

**Comment by grochocki** [comment]
I'll take a closer look this weekend to see if we can come up with a more robust solution here.

**Comment by artemious7** [comment]
Perhaps, you could simply add a setting to disable adding namespaces at all? I couldn't find such setting.

**Comment by grochocki** [comment]
Still debugging; wasn't able to come up with a good solution here yet.

**Comment by XiaotianNetlift** [comment]
@artemious7 The workaround works by modifying the xmlns to avoid namespace collision before formatting. Of course, you need to restore it to its original value once you've done formatting it, since `xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation+asdasd"` obviously doesnt make sense. This is the basis of #171 

Looking the extension on [vs marketplace](https://marketplace.visualstudio.com/items?itemName=TeamXavalon.XAMLStyler), it is last updated on 16/01/2019, 07:08:12. However. my fix was merged on march 17, 2019.

**Comment by XiaotianNetlift** [comment]
There is no such setting because that is how `XDocument` currently works:
```
// output: <asd:Example xmlns="asd" xmlns:asd="asd" />
System.Xml.Linq.XDocument.Parse("<Example xmlns=\"asd\" xmlns:asd=\"asd\"></Example>").ToString()
```
Of which we have no control.

**Comment by grochocki** [comment]
> Looking the extension on vs marketplace, it is last updated on 16/01/2019, 07:08:12. However. my fix was merged on march 17, 2019.

@XiaotianNetlift Reopened to track getting this workaround live _or_ finding an alternative solution. There are some tricky bugs being reported around reset settings, so trying to reduce churn as we work through those (though, I am not able to reproduce those issues myself).

**Comment by artemious7** [comment]
So what should I do to stop adding namespaces to all elements on each try to format? 
Should I build the extension from source and install it? Will it fix this issue? Or even then I would have to use the workaround?

**Comment by grochocki** [comment]
Should be addressed in [3.1912.9](https://github.com/Xavalon/XamlStyler/releases/tag/Release-3.1912.9), available now!
